src/test/java/org/dependencytrack/tasks/BomUploadProcessingTaskTest.java

```diff
@@ -221,4 +215,127 @@ public void informIssue2859Test() throws Exception {
                 .isThrownBy(() -> new BomUploadProcessingTask().inform(new BomUploadEvent(project.getUuid(), bomBytes)));
     }
 
+    @Test
+    public void informWithBomContainingLicenseExpressionTest() throws Exception {
+        final var project = new Project();
+        project.setName("acme-app");
+        qm.persist(project);
+
+        final byte[] bomBytes = """
+                {
+                  "bomFormat": "CycloneDX",
+                  "specVersion": "1.4",
+                  "serialNumber": "urn:uuid:3e671687-395b-41f5-a30f-a58921a69b79",
+                  "version": 1,
+                  "components": [
+                    {
+                      "type": "library",
+                      "publisher": "Acme Inc",
+                      "group": "com.acme",
+                      "name": "tomcat-catalina",
+                      "version": "9.0.14",
+                      "licenses": [
+                        {
+                          "expression": "EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0"
+                        }
+                      ]
+                    }
+                  ]
+                }
+                """.getBytes(StandardCharsets.UTF_8);
+
+        new BomUploadProcessingTask().inform(new BomUploadEvent(project.getUuid(), bomBytes));
+        assertConditionWithTimeout(() -> NOTIFICATIONS.size() >= 2, Duration.ofSeconds(5));
+
+        assertThat(qm.getAllComponents(project)).satisfiesExactly(component -> {
+            assertThat(component.getLicense()).isNull();
+            assertThat(component.getLicenseExpression()).isEqualTo("EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0");
+            assertThat(component.getResolvedLicense()).isNull();
+        });
+    }
+
+    @Test
+    public void informWithBomContainingLicenseExpressionWithSingleIdTest() throws Exception {
+        final var license = new License();
+        license.setLicenseId("EPL-2.0");
+        license.setName("Eclipse Public License 2.0");
+        qm.persist(license);
+
+        final var project = new Project();
+        project.setName("acme-app");
+        qm.persist(project);
+
+        final byte[] bomBytes = """
+                {
+                  "bomFormat": "CycloneDX",
```

Comment:
if this is a test to upload a bom that is of SPDX, should we include appropriate spdx nomenclature outside of just the license?  I suspect if someone uploads an SPDX bom, they would want that information to stay when downloading it too.

Reply:
SPDX BOMs are not supported at all right now; Uploads of such BOMs will be rejected. We have https://github.com/DependencyTrack/dependency-track/issues/1746 open, but doesn't look like we're going to address that anytime soon.

For this change, we just want to ensure that SPDX license expressions within a CDX BOM are correctly ingested.

Reply:
🤦‍♀️ for various reasons we discussed... i swear it was!  Thanks for bringing me back to reality 😺